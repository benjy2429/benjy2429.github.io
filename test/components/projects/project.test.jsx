import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Project from '../../../src/components/projects/project';

const mockProject = {
  title: 'Demo project',
  image: 'project-image.jpg',
  text: 'Project description'
};

const mockProjectWithLink = {
  ...mockProject,
  link: {
    text: 'Visit site',
    href: 'test.com'
  }
};

const mockProjectWithFile = {
  ...mockProject,
  file: 'file.pdf'
};

describe('Project component', () => {
  it('renders a project', () => {
    const component = shallow(<Project {...mockProject} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders a project with a link', () => {
    const component = shallow(<Project {...mockProjectWithLink} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders a project with a file', () => {
    const component = shallow(<Project {...mockProjectWithFile} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
