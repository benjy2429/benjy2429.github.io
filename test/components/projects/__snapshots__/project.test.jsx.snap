// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component renders a project 1`] = `
<article
  className="project"
>
  <img
    alt=""
    className="project-image"
    src="/assets/projects/project-image.jpg"
  />
  <div
    className="project-content"
  >
    <h3
      className="project-title"
    >
      Demo project
    </h3>
    <p
      className="project-description"
    >
      Project description
    </p>
  </div>
</article>
`;

exports[`Project component renders a project with a file 1`] = `
<article
  className="project"
>
  <img
    alt=""
    className="project-image"
    src="/assets/projects/project-image.jpg"
  />
  <div
    className="project-content"
  >
    <h3
      className="project-title"
    >
      Demo project
    </h3>
    <p
      className="project-description"
    >
      Project description
    </p>
    <a
      className="project-link"
      href="/download/file.pdf"
    >
      Download
    </a>
  </div>
</article>
`;

exports[`Project component renders a project with a link 1`] = `
<article
  className="project"
>
  <img
    alt=""
    className="project-image"
    src="/assets/projects/project-image.jpg"
  />
  <div
    className="project-content"
  >
    <h3
      className="project-title"
    >
      Demo project
    </h3>
    <p
      className="project-description"
    >
      Project description
    </p>
    <a
      className="project-link"
      href="test.com"
    >
      Visit site
    </a>
  </div>
</article>
`;
